torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,407 @@
+    from the World Database of Protected Areas (WDPA).
+
+    Dataset features:
+
+    * 1018 areas globally sampled from the WDPA
+    * 10-Band Sentinel-2
+    * Dual-pol Sentinel-1
+    * ESA WorldCover Land Cover
+    * Visible Infrared Imaging Radiometer Suite NightTime Day/Night Band
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://ieeexplore.ieee.org/document/10089830
+
+    .. versionadded:: 0.5
+    """
+
+    url = "https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/"
+
+    modality_urls = {
+        "esa_wc": {"esa_wc/ESA_WC.zip"},
+        "viirs": {"viirs/VIIRS.zip"},
+        "mask": {"mask/mask.zip"},
+        "s1": {"s1/s1_part1.zip", "s1/s1_part2.zip"},
+        "s2_temporal_subset": {
+            "s2_temporal_subset/s2_temporal_subset_part1.zip",
+            "s2_temporal_subset/s2_temporal_subset_part2.zip",
+        },
+        "s2_autumn": {"s2_autumn/s2_autumn_part1.zip", "s2_autumn/s2_autumn_part2.zip"},
+        "s2_spring": {"s2_spring/s2_spring_part1.zip", "s2_spring/s2_spring_part2.zip"},
+        "s2_summer": {"s2_summer/s2_summer_part1.zip", "s2_summer/s2_summer_part2.zip"},
+        "s2_winter": {"s2_winter/s2_winter_part1.zip", "s2_winter/s2_winter_part2.zip"},
+        "split_IDs": {"split_IDs/split_IDs.csv"},
+    }
+
+    md5s = {
+        "ESA_WC.zip": "72b2ee578fe10f0df85bdb7f19311c92",
+        "VIIRS.zip": "4eff014bae127fe536f8a5f17d89ecb4",
+        "mask.zip": "87c83a23a73998ad60d448d240b66225",
+        "s1_part1.zip": "d8a911f5c76b50eb0760b8f0047e4674",
+        "s1_part2.zip": "a30369d17c62d2af5aa52a4189590e3c",
+        "s2_temporal_subset_part1.zip": "78c2d05514458a036fe133f1e2f11d2a",
+        "s2_temporal_subset_part2.zip": "076cd3bd00eb5b7f5d80c9e0a0de0275",
+        "s2_autumn_part1.zip": "6ee7d1ac44b5107e3663636269aecf68",
+        "s2_autumn_part2.zip": "4fc5e1d5c772421dba553722433ac3b9",
+        "s2_spring_part1.zip": "2a89687d8fafa7fc7f5e641bfa97d472",
+        "s2_spring_part2.zip": "5845dcae0ab3cdc174b7c41edd4283a9",
+        "s2_summer_part1.zip": "73ca8291d3f4fb7533636220a816bb71",
+        "s2_summer_part2.zip": "5b5816bbd32987619bf72cde5cacd032",
+        "s2_winter_part1.zip": "ca958f7cd98e37cb59d6f3877573ee6d",
+        "s2_winter_part2.zip": "e7aacb0806d6d619b6abc408e6b09fdc",
+        "split_IDs.csv": "cb5c6c073702acee23544e1e6fe5856f",
+    }
+
+    mask_cmap = {1: (0, 153, 0), 0: (255, 255, 255)}
+
+    wc_cmap = {
+        10: (0, 160, 0),
+        20: (150, 100, 0),
+        30: (255, 180, 0),
+        40: (255, 255, 100),
+        50: (195, 20, 0),
+        60: (255, 245, 215),
+        70: (255, 255, 255),
+        80: (0, 70, 200),
+        90: (0, 220, 130),
+        95: (0, 150, 120),
+        100: (255, 235, 175),
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        modality: list[str] = ["mask", "esa_wc", "viirs", "s2_summer"],
```

Comment:
```suggestion
        modality: Sequence[str] = ["mask", "esa_wc", "viirs", "s2_summer"],
```
This can be imported from typing at the top of the file. This will allow people to use either a list or tuple if they want to.

Reply:
I do append on the `modality` and that does not work with tuples. Doing a type changing in the init but it does not seem to work. Any suggestions?

Reply:
Ah, if append is required then it has to be a list. Let's keep it as is.